### PR TITLE
Upgrade to Python 3

### PR DESCRIPTION
**Pull request by sean-smith**
* Remove support for Python 2.
* Add Python 3 support

Signed-off-by: Sean Smith <seaam@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -30,18 +30,15 @@ def read(fname):
...
-    requires.append("cryptography==2.1.4")  # dependency of paramiko
-    requires.append("pycparser==2.18")
-else:
-    requires.append("paramiko>=2.4.2")
+requires = [
+    "requests>=2.21.0",
+    "boto3>=1.7.55",
+    "python-dateutil>=2.6.1",
+    "retrying>=1.3.3",
+    "future>=0.17.1",
```
Comment (demartinofra) [inline setup.py]: we can probably drop this

File: setup.py [hunk 2]
```diff
@@ -30,18 +30,15 @@ def read(fname):
...
-    # For python2.6 we have to require argparse since it
-    # was not in stdlib until 2.7.
-    requires.append("argparse>=1.4")
-    requires.append("idna==2.6")
-    requires.append("paramiko==2.3.3")
-    requires.append("cryptography==2.1.4")  # dependency of paramiko
-    requires.append("pycparser==2.18")
-else:
-    requires.append("paramiko>=2.4.2")
+requires = [
```
Comment (demartinofra) [inline setup.py]: shall we sync the requirements.txt file with this list?

File: setup.py [hunk 3]
```diff
@@ -30,18 +30,15 @@ def read(fname):
...
+    "requests>=2.21.0",
+    "boto3>=1.7.55",
+    "python-dateutil>=2.6.1",
+    "retrying>=1.3.3",
+    "future>=0.17.1",
+    "configparser>=3.7.4",
+    "paramiko>=2.4.2",
+]
 
 setup(
```
Comment (demartinofra) [inline setup.py]: shall we add `python_requires='>=3.4'`?

https://packaging.python.org/guides/dropping-older-python-versions/

File: setup.py (hunk 1 above)
Comment (sean-smith) [reply setup.py]: done

File: setup.py (hunk 2 above)
Comment (sean-smith) [reply setup.py]: done

File: setup.py (hunk 3 above)
Comment (sean-smith) [reply setup.py]: added

File: setup.py (hunk 1 above)
Comment (sean-smith) [reply setup.py]: removed

**Comment by sean-smith** [comment]
> Also I think we can remove all the future usages (e.g. future.moves...)

done

**Pull request merged**